core/domain/email_manager.py

```diff
@@ -164,5 +178,71 @@ def send_post_signup_email(user_id):
         EMAIL_FOOTER.value)
 
     _send_email(
-        user_id, feconf.SYSTEM_COMMITTER_ID, email_models.INTENT_SIGNUP,
+        user_id, feconf.SYSTEM_COMMITTER_ID, feconf.EMAIL_INTENT_SIGNUP,
         email_subject, email_body)
+
+
+def require_valid_intent(intent):
+    if intent not in feconf.VALID_MODERATOR_ACTIONS:
+        raise Exception('Unrecognized email intent: %s' % intent)
+
+
+def _get_email_config(intent):
```

Comment:
Technically, this is the case in these examples: the intent is a subset of the action. What I mean by that is, the action of unpublishing an exploration consists of two things: changing the exploration's status, and sending an email. The intent refers only to the email. Perhaps it might not have been a good idea to use the same constant strings for these things, but I think it's really the names of the constants that matter so I'm not _too_ worried.

You could imagine a future scenario where moderators had a web form that allowed them to essentially send a message to any contributor. They would be able to choose from a list of intents (which makes searching past emails easier in the future), and compose an arbitrary email body. The action here is basically 'send private message'; the intent is whatever the moderator picks.